tests/encodeinput.cpp

```diff
@@ -167,40 +167,49 @@ EncodeOutput::~EncodeOutput()
         fclose(m_fp);
 }
 
-EncodeOutput* EncodeOutput::create(const char* outputFileName, int width , int height)
+EncodeOutput* EncodeOutput::create(const char* outputFileName, int width,
+                                   int height, const char* codecName)
 {
-    EncodeOutput * output = NULL;
-    if(outputFileName==NULL)
+    EncodeOutput* output = NULL;
+    if (outputFileName == NULL)
         return NULL;
-    const char *ext = strrchr(outputFileName,'.');
-    if(ext==NULL)
+    const char* ext = strrchr(outputFileName, '.');
+    std::string codec(""); // needed for strcasecmp to work
+    if (!ext && !codecName)
         return NULL;
-    ext++;//h264;264;jsv;avc;26l;jvt;ivf
-    if(strcasecmp(ext,"h264")==0 ||
-        strcasecmp(ext,"264")==0 ||
-        strcasecmp(ext,"jsv")==0 ||
-        strcasecmp(ext,"avc")==0 ||
-        strcasecmp(ext,"26l")==0 ||
-        strcasecmp(ext,"jvt")==0 ) {
-            output = new EncodeOutputH264();
+    ext++;
+
+    if (codecName)
+        codec.assign(codecName);
+
+    if (!strcasecmp("AVC", codec.c_str()) || !strcasecmp(ext, "h264")
+        || !strcasecmp(ext, "264") || !strcasecmp(ext, "jsv")
+        || !strcasecmp(ext, "avc") || !strcasecmp(ext, "26l")
+        || !strcasecmp(ext, "jvt")) {
+        output = new EncodeOutputH264();
+    }
+    else if (!strcasecmp("VP8", codec.c_str())
+             || (!strcasecmp("VP8", codec.c_str()) && !strcasecmp(ext, "ivf"))
```

Comment:
Hi @lizhong1008 :
This will introduce unnecessary complexity to our encoder codec guess logical. You can look at EncodeOutput::create. We only guess from extension yet. Actually for Daniel’s patch, you can add `-c vp9` to get vp9 encoded .ivf file. Daniel and I do not think ivf should for vp8 only.
The main difference between Daniel and me is I think we should treat vp8 as default codec format for ivf. i.e. 
1.	If you use `yamiencode –i xxx.264 –o ttt.ivf`, we will encode vp8 for you. 
2.	If you use `yamiencode –i xxx.264 –c vp9 –o ttt.ivf`, we will encode vp9 for you.
Daniel suggest we failed and report an error to user in case 1. 
What do you prefer?